web/src/app/alerts/AlertsList.js

```diff
@@ -0,0 +1,369 @@
+  const favoritesQueryStatus = useQuery(
+    gql`
+      query($input: ServiceSearchOptions) {
+        services(input: $input) {
+          nodes {
+            id
+          }
+        }
+      }
+    `,
+    {
+      variables: {
+        favoritesOnly: true,
+        first: 1,
+      },
+    },
+  )
+
+  // checks to show no favorites warning
+  const noFavorites = !favoritesQueryStatus.data?.nodes?.length
+  const showNoFavoritesWarning =
+    _showNoFavoritesWarning &&
+    !allServices &&
+    !props.serviceID &&
+    !isFirstLogin &&
+    noFavorites
+
+  /*
+   * Closes the no favorites warning snackbar only if clicking
+   * away to lose focus
+   */
+  function handleCloseNoFavoritesWarning(event, reason) {
+    if (reason === 'clickaway') {
+      setShowNoFavoritesWarning(false)
+    }
+  }
+
+  // alerts list query variables
+  const variables = {
+    input: {
+      filterByStatus: getStatusFilter(filter),
+      first: 25,
+      // default to favorites only, unless viewing alerts from a service's page
+      favoritesOnly: !props.serviceID && !allServices,
+    },
+  }
+
+  if (props.serviceID) {
+    variables.input.filterByServiceID = [props.serviceID]
+  }
+
+  // checkbox action mutations
+  const [ackAlerts] = useMutation(updateMutation)
+  const [closeAlerts] = useMutation(updateMutation)
+  const [escalateAlerts] = useMutation(escalateMutation)
+
+  /*
+   * Called on the successful completion of a checkbox action.
+   * Sets showing the update snackbar and sets the message to be
+   * displayed
+   */
+  function onCompleted(numUpdated, checkedItems) {
+    setUpdateComplete(true) // for create fab transition
+    setUpdateMessage(`${numUpdated} of ${checkedItems.length} alerts updated`)
+  }
+
+  /*
+   * Adds border of color depending on each alert's status
+   * on left side of each list item
+   */
+  function getListItemStatus(s) {
+    switch (s) {
+      case 'StatusAcknowledged':
+        return 'warn'
+      case 'StatusUnacknowledged':
+        return 'err'
+      case 'StatusClosed':
+        return 'ok'
+    }
+  }
+
+  /*
+   * Passes the proper actions to ListControls depending
+   * on which tab is currently filtering the alerts list
+   */
+  function getActions() {
+    const actions = []
+
+    if (filter !== 'closed' && filter !== 'acknowledged') {
+      actions.push({
+        icon: <AcknowledgeIcon />,
+        label: 'Acknowledge',
+        ariaLabel: 'Acknowledge Selected Alerts',
+        onClick: checkedItems => {
+          ackAlerts({
+            variables: {
+              input: {
+                alertIDs: checkedItems,
+                newStatus: 'StatusAcknowledged',
+              },
+            },
+          })
+            .then(res =>
+              onCompleted(res?.data?.updateAlerts?.length ?? 0, checkedItems),
+            )
+            .catch(err => {
+              setUpdateComplete(true)
+              setErrorMessage(err.message)
+            })
```

Comment:
As above, let's move away from relying on the Promise API for these, since we are provided state by the `useMutation` hook

Reply:
`onCompleted` is not properly called from within the `useMutation` options object when calling the `mutate` func from the hook, so we are forced to use the Promise API in this situation

Reply:
I'm not sure we need to, `data.updatedAlerts.length` should be provided by `useMutation` as well as that the mutation was complete. Both of which we are tracking in state, but should be relying on the hooks API instead.

The only value we need to persist is the total number of alerts, which we have when the button is clicked, before we call the mutation. (and that's assuming we don't have access to the mutations input vars from the hooks API)

Reply:
That's correct, the main reason I moved the onComplete/onError calls to then/catch, respectively, is because `checkedItems` is not persisted and we don't have access to it outside of the `onClick()` provided to us by `QueryList` and children.

Reply:
In that case we can store that single value in state, instead of the others and still drop the usage of the Promise API.

Maybe call it `selectedAlertCount` or `lastSelectedAlertCount`?

We already have `result.error` for the error state and completed & count based on `data` (or `.called && !.loading`).

Reply:
That works, though I'm not sure how far in which direction we want to go since we're at a win/lose in either way: data in state that could be avoided as it is available elsewhere, or, using a promise to workaround an apollo bug